bitbybit/README.md

```diff
@@ -150,6 +150,27 @@ const A: Bitfield1 = Bitfield1::DEFAULT;
 
 Default values are used as-is, even if they affect bits that aren't defined within the bitfield.
 
+## Setting all fields at once using the builder syntax (experimental)
+
+It is possible to set all fields at once, like this:
+```rs
+const T: Test = Test::builder()
+    .with_baudrate(0x12)
+    .with_some_other_bits(u4::new(0x2))
+    .with_array([1, 2, 3, 4])
+    .build();
+```
+
+Using `builder()` it is impossible to forget setting any fields. This is checked at compile time: If any field is not set, `build()` can not be called.
+
+At the moment, it is required to set all fields in the same order as they are specified. As Rust's const generics become more powerful, this restriction might be lifted.
+
+For the `builder()` to be available, the following has to be true:
+- The bitfield has to be completely filled with writable fields (no gaps) OR there has to be a default value specified,
```

Comment:
For a register with a mix of read-only and read-write fields, perhaps it would make sense to allow constructing a builder from an existing value, instead of the compile-time default.

Reply:
Good idea! We could allow this as a function while the mask is still 0. Something like this:

```
let a = A::builder()
  .with_overwritten_raw_value(0x12345678)
  .with_foo(false)
  .with_bar(false)
  .build();
```

Though I can't quite think of a good name for this function